### PR TITLE
First commit. Create Security config, filters, auth controller and se…

### PR DESCRIPTION
**Pull request by aldogayala14**
Adding Spring Security config, filters, jwt utils, auth controller and service. Authorize requests permit path: "/auth/*" and "/rol".
Rename in model User class, new name "User Entity". Revision in relation with model/Role.

### CODE REVIEW

File: src/main/java/com/alkemy/wallet/security/dto/UserAuthDto.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.alkemy.wallet.security.dto;
+
+import lombok.Data;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.Size;
+
+@Data
+public class UserAuthDto {
```
Comment (estebanfak) [inline src/main/java/com/alkemy/wallet/security/dto/UserAuthDto.java:9]: Donde usamos éste Dto?

**Review by estebanfak — APPROVED** [review]

**Review by estebanfak — APPROVED** [review]

File: src/main/java/com/alkemy/wallet/security/dto/UserAuthDto.java (hunk 1 above)
Comment (aldogayala14) [reply src/main/java/com/alkemy/wallet/security/dto/UserAuthDto.java:9]: Hay que eliminar este Dto, se uso UserDto en el directorio padre.

**Comment by gianfrancoodp** [comment]
Laburazo!

**Review by gianfrancoodp — APPROVED** [review]

**Review by NicoRM11 — APPROVED** [review]

**Review by DeniseColotta — COMMENTED** [review]
Muy bueno

**Review by DeniseColotta — APPROVED** [review]

**Review by gianfrancoodp — APPROVED** [review]

**Pull request merged**